src/runtime/gc_stack_portable.go

```diff
@@ -61,3 +55,12 @@ func trackPointer(ptr unsafe.Pointer)
 func swapStackChain(dst **stackChainObject) {
 	*dst, stackChainStart = stackChainStart, *dst
 }
+
+func init() {
```

Comment:
Since custom GC won't call markStack it gets removed during compilation as does the volatility. An init method is never removed so seems to allow the volatility to be correct regardless of markStack.

Reply:
I agree at the moment it's not removed, I'm just concerned about smarter optimizers that notice that all the writes to `stackChainStart` happen after the load..

Maybe we should keep the volatile load in `markStack` and add one in `init` as you have here?

Reply:
(Edit: tested this with the standard conservative gc and it seems to work.)

Reply:
That's a good point. I went ahead and added it to both locations for now

Reply:
Don't put it in an init. As I said in the comment above, the following is unsafe:

> Since custom GC won't call markStack

Therefore, because `markStack` must be used, this change can be reverted.